chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -559,14 +621,50 @@ where
         let input0_getter =
             || tx.inputs().get(0).ok_or(ConnectTransactionError::MissingOutputOrSpent);
 
+        let attempt_to_spend_stake = tx
+            .inputs()
+            .iter()
+            .map(|i| {
+                self.utxo_cache
+                    .utxo(i.outpoint())
+                    .ok_or(ConnectTransactionError::MissingOutputOrSpent)
+            })
+            .collect::<Result<Vec<_>, _>>()?
+            .iter()
+            .any(|utxo| match utxo.output().purpose() {
+                OutputPurpose::Transfer(_)
+                | OutputPurpose::LockThenTransfer(_, _)
+                | OutputPurpose::Burn => false,
+                OutputPurpose::StakePool(_) | OutputPurpose::SpendStakePool(_) => true,
+            });
+        ensure!(
+            !attempt_to_spend_stake,
+            ConnectTransactionError::AttemptToSpendStakedCoins
+        );
+
+        let attempt_to_use_spend_stake_pool =
+            tx.outputs().iter().any(|output| match output.purpose() {
+                OutputPurpose::Transfer(_)
+                | OutputPurpose::LockThenTransfer(_, _)
+                | OutputPurpose::Burn
+                | OutputPurpose::StakePool(_) => false,
+                OutputPurpose::SpendStakePool(_) => true,
+            });
+
+        ensure!(
+            !attempt_to_use_spend_stake_pool,
+            ConnectTransactionError::AttemptToSpendStakedCoins
+        );
+
```

Comment:
Where's the part where we increase the balance of the staking pool?

There's also the question of distributing rewards. We should have a discussion about this.

Reply:
It's not there. I realized how the reward should be distributed only yesterday after a discussion with you. Will have to add it

Reply:
done, for now all reward goes to a pool